src/Microsoft.SqlTools.Hosting/Hosting/Protocol/MessageDispatcher.cs

```diff
@@ -324,13 +354,15 @@ private async Task ListenForMessages(CancellationToken cancellationToken)
 
             if (handlerToAwait != null)
             {
-                if (this.ParallelMessageProcessing)
+                if (this.ParallelMessageProcessing && isParallelProcessingSupported)
```

Comment:
This currently means that any non-parallel enabled messages will block all other messages from being processed until it's completed. But that means we're only getting some of the benefit of going parallel in the first place - ideally we should be having all parallel messages kicked off as soon as they're read in while any non-parallel ones will be handled sequentially.

Reply:
Yes we are only getting part of the benefit here. I attempted to get msg loop on its own thread but that will be a big change so we are going this route for now.